### PR TITLE
Improve JUnit tests

### PR DESCRIPTION
**Pull request by HomelessSkittle**
(Hopefully) Improved code coverage by expanding JUnit tests for:
 - EntityTest.java
 - Relationship.java
Added JUnit tests for:
 - Formal.java

(Mostly doing this just to see if code coverage actually improved, since I think it only runs on open PR's.)

### CODE REVIEW

**Review by SerbianSensation — APPROVED** [review]
Improving code coverage might be a good thing 👍

**Comment by HomelessSkittle** [comment]
Gonna wait until the meeting at 4 to merge this - Not 100% sure what the changes Phil made were, or how they do/do not work with these

**Pull request merged**
